server/src/main/java/org/opensearch/rest/action/admin/cluster/RestRestoreRemoteStoreAction.java

```diff
@@ -45,6 +45,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         );
         restoreRemoteStoreRequest.waitForCompletion(request.paramAsBoolean("wait_for_completion", false));
         restoreRemoteStoreRequest.restoreAllShards(request.paramAsBoolean("restore_all_shards", false));
+        restoreRemoteStoreRequest.clusterUUID(request.param("clusterUUID", ""));
```

Comment:
Why do we need to take this in the request. Shouldn't the remote repository serve as a custodian for this value?

Reply:
Agree this should be provided by remote but for phase we ask this from user. putting clusterUUID in remote will be done in later phases. There we some edge cases which we need to think about if we want to fetch clusterUUID from remote. This made more sense when we also think about auto restore/recovery. Let me know your thoughts

Reply:
+1 to @linuxpi, In the first phase, to keep things simple we are building the foundation blocks to upload/ download the metadata to remote store. In case of disaster recovery, post Quorum Loss recovery the user is expected to restore the metadata state. The user has full control from which clusterUUID they wish to restore the metadata. Incrementally in later phases, this will be done automatically by finding the best state (clusterUUID) from remote state.